src/ast/irbuilderbpf.h

```diff
@@ -183,6 +183,14 @@ class IRBuilderBPF : public IRBuilder<>
                        Value *data_len,
                        const location &loc);
 
+  // For a type T, creates an integer expression representing the byte offset
+  // of the element at the given index in T[]. Used for array dereferences and
+  // pointer arithmetic.
+  llvm::Value *CreatePtrOffset(
+      const SizedType &type,
+      llvm::Value *index,
+      std::optional<AddrSpace> addrSpace = std::nullopt);
```

Comment:
Any reason to make this parameter optional when it's always being passed in? Unless it's needed, I'd do without it for simplicity and let someone make it optional in the future if it's useful